src/main/java/io/zold/api/WalletsIn.java

```diff
@@ -98,10 +121,20 @@ public Wallet create() throws IOException {
         final Path wpth = this.path.value().resolve(
             new JoinedText(
                 ".",
-                Long.toHexString(new Random().nextLong()),
+                Long.toHexString(this.random.nextLong()),
                 this.ext
             ).asString()
         );
+        if (Files.exists(wpth)) {
```

Comment:
@Vatavuk let's use `wpth.toFile().exists()`. See [here](https://rules.sonarsource.com/java/tag/performance/RSPEC-3725) for more details